### PR TITLE
Fogl 1366: Convert the north Http plugin to Async

### PR DESCRIPTION
**Pull request by sstefano69**

### CODE REVIEW

**Comment by praveen-garg** [comment]
@sstefano69  Please fix the tests for this specific plugin too; 
/cc @ashish-jabble  please review; as this is broken on develop.

File: python/foglamp/plugins/north/http_north/http_north.py [hunk 1]
```diff
@@ -75,8 +75,13 @@ def plugin_init(data):
...
 
-def plugin_send(data, payload, stream_id):
-    return http_north.send_payloads(payload, stream_id)
+async def plugin_send(data, payload, stream_id):
+
+    is_data_sent, new_last_object_id, num_sent = await http_north.send_payloads(payload, stream_id)
+
+    return is_data_sent, new_last_object_id, num_sent
+
+
```
Comment (ashish-jabble) [inline python/foglamp/plugins/north/http_north/http_north.py]: Remove above spaces

File: python/foglamp/plugins/north/http_north/http_north.py [hunk 2]
```diff
@@ -75,8 +75,13 @@ def plugin_init(data):
     return config
 
 
-def plugin_send(data, payload, stream_id):
-    return http_north.send_payloads(payload, stream_id)
+async def plugin_send(data, payload, stream_id):
+
```
Comment (ashish-jabble) [inline python/foglamp/plugins/north/http_north/http_north.py]: Do we need `data`, as it seems to be unused? And remove space line.

File: python/foglamp/plugins/north/http_north/http_north.py (hunk 1 above)
Comment (sstefano69) [reply python/foglamp/plugins/north/http_north/http_north.py]: done

File: python/foglamp/plugins/north/http_north/http_north.py (hunk 2 above)
Comment (sstefano69) [reply python/foglamp/plugins/north/http_north/http_north.py]: it is needed, 
the sign of the plugin_send call should be the same for every plugin.

**Pull request merged**